### PR TITLE
chore: add prepare script for @yext/pages

### PR DESCRIPTION
**Pull request by mkilpatrick**

### CODE REVIEW

File: packages/pages/package.json [hunk 1]
```diff
@@ -50,7 +50,8 @@
     "generate-notices": "generate-license-file --input package.json --output ./THIRD-PARTY-NOTICES --overwrite",
     "prepublishOnly": "pnpm run build:release",
     "storybook": "start-storybook -p 6006",
-    "build-storybook": "build-storybook"
+    "build-storybook": "build-storybook",
+    "prepare": "pnpm run build:release"
```
Comment (mancusi) [inline packages/pages/package.json:54]: we should rename the build step to `build:prepare` if we want to run it on the prepare hook.

**Review by mancusi — CHANGES_REQUESTED** [review]

File: packages/pages/package.json (hunk 1 above)
Comment (mkilpatrick) [reply packages/pages/package.json:54]: It's also used in `prepublishOnly` above. You still want to rename to `build:prepare`?

**Review by mancusi — APPROVED** [review]

File: packages/pages/package.json [hunk 2]
```diff
@@ -36,9 +36,9 @@
   },
   "scripts": {
     "watch": "node src/bundler --watch",
-    "build": "node src/bundler.js",
-    "build:release": "pnpm run build && pnpm run types",
-    "types": "tsc",
+    "build:js": "node src/bundler.js",
+    "build:types": "tsc",
+    "build": "pnpm run build:js && pnpm run build:types",
```
Comment (mancusi) [inline packages/pages/package.json:41]:
```suggestion
    "build": "pnpm build:js && pnpm build:types",
```

File: packages/pages/package.json [hunk 3]
```diff
@@ -48,9 +48,9 @@
     "docs:serve": "vitepress serve docs --port 8080",
     "test": "jest",
     "generate-notices": "generate-license-file --input package.json --output ./THIRD-PARTY-NOTICES --overwrite",
-    "prepublishOnly": "pnpm run build:release",
     "storybook": "start-storybook -p 6006",
-    "build-storybook": "build-storybook"
+    "build-storybook": "build-storybook",
+    "prepare": "pnpm run build"
```
Comment (mancusi) [inline packages/pages/package.json:53]: similar comment.

**Review by mancusi — CHANGES_REQUESTED** [review]

**Review by mancusi — APPROVED** [review]

**Review by benmcginnis — COMMENTED** [review]
lgtm, though I am not an expert. Does it make local development work when referencing a branch?

**Comment by mkilpatrick** [comment]
As in referencing a GH url? It should.

**Pull request merged**